app/javascript/app/pages/national-context/sectoral-activity/sectoral-activity-selectors.js

```diff
@@ -0,0 +1,418 @@
+    } else {
+      filteredData = isArray(emissionActivities) &&
+        selectedIndicator &&
+        emissionActivities.filter(
+          emissionData => emissionData.sector === selectedIndicator.label
+        );
+    }
+
+    return { data: filteredData };
+  }
+);
+
+const SECTION_COLORS = {
+  Waste: '#06214C',
+  Agriculture: '#FC7E4B',
+  Energy: '#2EC9DF',
+  Forestry: '#FCA683',
+  'Industrial Process and Product Use': '#FFC735',
+  'no-data': '#ffffff'
+};
+
+const COLORS = [ '#06214C', '#FC7E4B', '#2EC9DF', '#FCA683', '#FFC735' ];
+
+const getColorsForActivities = activities => {
+  const mapColors = {};
+  activities.forEach((a, i) => {
+    mapColors[a] = COLORS[i];
+  });
+  return mapColors;
+};
+
+const countryStyles = color => ({
+  default: {
+    fill: color,
+    fillOpacity: 1,
+    stroke: '#ffffff',
+    strokeWidth: 0.2,
+    outline: 'none'
+  },
+  hover: { fill: color, stroke: '#ffffff', strokeWidth: 0.6, outline: 'none' },
+  pressed: { fill: color, stroke: '#ffffff', strokeWidth: 0.2, outline: 'none' }
+});
+
+const getPathsWithStylesForAdaptation = (
+  adaptationData,
+  selectedOptions,
+  sectors
+) =>
+  {
+    if (!adaptationData || !selectedOptions || !sectors) return null;
+
+    const provincesData = adaptationData.values &&
+      adaptationData.values.filter(o => o.indicator_code === ADAPTATION_CODE);
+    const groupedByProvince = provincesData &&
+      groupBy(provincesData, 'location_iso_code3');
+
+    const map = {};
+    if (groupedByProvince) {
+      Object.keys(groupedByProvince).forEach(province => {
+        map[province] = groupedByProvince[province][0] &&
+          groupedByProvince[province][0].values &&
+          groupedByProvince[province][0].values[0] &&
+          groupedByProvince[province][0].values[0].value;
+      });
+    }
+
+    const YES_NO_COLORS = { no: '#001880', yes: '#2EC9DF' };
+
+    const paths = [];
+    const legend = [];
+
+    indonesiaPaths.forEach(path => {
+      const iso = path.properties && path.properties.code_hasc;
+      const province = map[iso];
+      if (province) {
+        const value = map[iso];
+        const color = YES_NO_COLORS[toLower(value)];
+
+        // Fill the legend
+        if (!legend.find(i => i.name === value)) {
+          legend.push({ name: value, color });
+        }
+        if (!legend.find(i => toLower(i.name) === toLower(NO_DATA_LEGEND))) {
+          legend.push({ name: NO_DATA_LEGEND, color: SECTION_COLORS[NO_DATA] });
+        }
+
+        const enhancedPath = {
+          ...path,
+          properties: { ...path.properties, tooltipValue: value }
+        };
+        paths.push({ ...enhancedPath, style: countryStyles(color) });
+      } else {
+        paths.push({ ...path, style: countryStyles(SECTION_COLORS[NO_DATA]) });
+      }
+    });
+
+    return { paths, legend };
+  };
+
+export const getPathsWithStyles = (provincesData, selectedOptions, sectors) => {
```

Comment:
It's a really long selector, maybe you could extract blocks, especially `Object.keys...` to separates methods, it will be more readable when block like this will be named method